### PR TITLE
Switch to Pytest

### PR DESCRIPTION
**Pull request by Ayub-Khan**

### CODE REVIEW

File: requirements/test.in [hunk 1]
```diff
@@ -1,11 +1,13 @@
 # Requirements for test runs
 
--r base.txt                         # Package dependencies
+-r base.in                         # Package dependencies
+-r travis.in
```
Comment (jmbowman) [inline requirements/test.in]: Why was this needed here?  We usually only need the dependencies from that file in CI and dev environments.

File: tox.ini [hunk 2]
```diff
@@ -31,15 +22,14 @@ commands =
     django20: pip install Django>=2.0,<2.1
     django21: pip install Django>=2.1,<2.2
     django22: pip install Django>=2.2,<2.3
-    python ./run_tests.py
+    python -Wd -m pytest {posargs}
```
Comment (jmbowman) [inline tox.ini]: Can we delete `run_tests.py` now?

File: requirements/test.in (hunk 1 above)
Comment (Ayub-Khan) [reply requirements/test.in]: travis includes tox-battery, was having some issues locally that dependencies were not being updated in tox env properly so added travis requirements here.

Comment (Ayub-Khan) [reply requirements/test.in]: removed.

**Comment by Ayub-Khan** [comment]
@jmbowman kindly review again.

**Pull request merged**